mitxgraders/baseclasses.py

```diff
@@ -205,6 +207,60 @@ def log_output(self):
         content = "\n".join(self.debuglog)
         return "<pre>{content}</pre>".format(content=content)
 
+    @classmethod
+    def ensure_text_inputs(cls, student_input, allow_lists=True, allow_single=True):
+        """
+        Ensures that student_input is a list of text strings.
```

Comment:
Can't it just be a single string too (not wrapped in a list)?

Reply:
Yes, thanks.

I briefly considered having AbstractGrader demand a list of strings, not string or list of strings.

That feels inappropriate for AbstractGrader, but all single-input graders inherit from ItemGrader, right? No! I had forgotten that IntegralGrader can be multi-input or single-input.